armi/reactor/converters/tests/test_axialExpansionChanger.py

```diff
@@ -555,6 +556,46 @@ def test_isFuelLocked(self):
             self.assertEqual(the_exception.error_code, 3)
 
 
+class TestInputHeightsConsideredHot(unittest.TestCase):
+    """verify thermal expansion for process loading of core"""
+
+    def setUp(self):
+        """provide the base case"""
+        _o, r = loadTestReactor(
+            os.path.join(TEST_ROOT, "detailedAxialExpansion"),
+            {"inputHeightsConsideredHot": True},
+        )
+        self.stdAssems = [a for a in r.core.getAssemblies()]
+
+    def test_coldHeight(self):
```

Comment:
So, the total height of the assemblies doesn't change if we enter cold inputs. But the height of every block inside it will?

I would have expect only certain blocks to grow in height, and ditto the assemblies.

I'm not _trying_ to be obtuse, I swear. But maybe a one-liner comment before each `for`-loop, that says what the `assert` statements are trying to prove?

Reply:
Love the PR.

I only have this one question about what I'm supposed to understand from this test.  Then we can merge it, sure.

Reply:
> So, the total height of the assemblies doesn't change if we enter cold inputs. But the height of every block inside it will?

Yes, exactly. In the blueprints defined in `armi.tests.detailedAxialExpansion`, the overall height of the assembly is preserved through the use of the "dummy" block.
```yaml
    dummy : &block_dummy
        coolant:
            shape: Hexagon
            material: Sodium
            Tinput: 25.0
            Thot: 450.0
            ip: 0.0
            mult: 1.0
            op: 16.75
```
So in the standard case, 
```python
self.stdAssems = [a for a in r.core.getAssemblies()]
```
_most_ of the blocks are an even 25.0 cm (the control assemblies have some differences to eventually stress test uniformMesh.py). And in the expanded case,
```python
testAssems = [a for a in rCold.core.getAssemblies()]
```
all of the blocks (except the dummy) are expanded based on the DeltaT between `Tinput` and `Thot`. The dummy block is then "chopped" to account for the accumulated change in height in the blocks below (this happens in armi.reactor.converters.axialExpansionChanger). So assuming that `Thot` > `Tinput` everywhere (which it is in `armi.tests.detailedAxialExpansion`), all of the blocks will be different heights (the specific height will vary on the temperature difference and expansion coefficients of the materials) but the overall assembly height will be preserved.

Reply:
@john-science , a few comments added to the docstring.